File: UnitTestNestjs/src/user/user.controller.ts
import { Body, Controller, Get, Param, Post, ValidationPipe } from "@nestjs/common";
import { UserService } from "./user.service";
import { UserCreateDto } from "./user.create.dto";

@Controller("user")
export class UserController {
    constructor(private userService: UserService) { }
    @Get("all")
    all() {
        return this.userService.getUsers()
    }

    @Post("create")
    async create(@Body(new ValidationPipe()) user: UserCreateDto) {
        return await this.userService.createUser(user)
    }
}

File: UnitTestSpringBoot/src/main/java/UnitTestSpringBoot/User/Controller/UserController.java
package UnitTestSpringBoot.User.Controller;

import UnitTestSpringBoot.User.Entity.DTO.UserCreateDTO;
import UnitTestSpringBoot.User.Entity.User;
import UnitTestSpringBoot.User.Service.UserService;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("user")
public class UserController {
    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("all")
    public List<User> getAll(){
       return userService.getAll();
    }
    @PostMapping("create")
    public User create(@RequestBody UserCreateDTO userCreateDTO){
        return userService.create(userCreateDTO);
    }
}
